src/layers.jl

```diff
@@ -127,18 +189,32 @@ end
 @functor Patching
 
 """
-    mlpblock(planes, expansion_factor = 4, dropout = 0., dense = Dense)
+    PosEmbedding{T}
 
-Feedforward block used in many vision transformer-like models.
+Positional embedding layer used by many vision transformer-like models. Instantiated with an 
+embedding vector which is a learnable parameter.
+"""
+struct PosEmbedding{T}
+  embedding_vector::T
+end
+
```

Comment:
There's no need to pad a singleton dimension with broadcasting.

```suggestion

PosEmbedding(embedsize, npatches; init = (dims) -> rand(Float32, dims)) =
  PosEmbedding(init(embedsize, npatches))

```